client/lib/query-manager/media/index.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * External dependencies
+ */
+import moment from 'moment';
+import { every, includes } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import PaginatedQueryManager from '../paginated';
+import MediaQueryKey from './key';
+import { DEFAULT_MEDIA_QUERY } from './constants';
+
+/**
+ * MediaQueryManager manages media which can be queried and change over time
+ */
+export default class MediaQueryManager extends PaginatedQueryManager {
+	/**
+	 * Returns true if the media item matches the given query, or false
+	 * otherwise.
+	 *
+	 * @param  {Object}  query Query object
+	 * @param  {Object}  media Item to consider
+	 * @return {Boolean}       Whether media item matches query
+	 */
+	matches( query, media ) {
+		return every( { ...DEFAULT_MEDIA_QUERY, ...query }, ( value, key ) => {
+			switch ( key ) {
+				case 'search':
+					if ( ! value ) {
```

Comment:
In addition to returning `true` for an empty search value, this will also happen if the user inputs `0` or any other falsy value. We might want to alter this check to:

```
if ( value !== '' ) {
```